casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -151,13 +151,31 @@ final class Estimator[F[_]: Sync: Log: Metrics: Span](
       } yield result
 
     // TODO: Since map scores are additive it should be possible to do this in parallel
-    latestMessagesHashes.toList.foldLeftM(Map.empty[BlockHash, Long]) {
+    /*latestMessagesHashes.toList.foldLeftM(Map.empty[BlockHash, Long]) {
       case (acc, (validator: Validator, latestBlockHash: BlockHash)) =>
         addValidatorWeightDownSupportingChain(
           acc,
           validator,
           latestBlockHash
         )
+    }*/
+
+    // Initial code above that commented out is slow when there is old latest message.
+    // The below solution is a temp solution until we have efficient way to calculate LCA.
+    latestMessagesHashes.toList.foldLeftM(Map.empty[BlockHash, Long]) {
+      case (acc, (validator: Validator, latestBlockHash: BlockHash)) =>
+        for {
+          topBlockNumber           <- blockDag.latestBlockNumber
+          latestMessageBlockNumber <- blockDag.lookup(latestBlockHash).map(_.get.blockNum)
+          r <- if (topBlockNumber - latestMessageBlockNumber > 1000)
```

Comment:
I like this fix better than the before one~! Good work. 
However, I am still picky on this hard-coded `1000`. :)
I saw this `1000` is used in multiple place. We'd better define a `val` currently and use that value in one place for now.